File: code/09-string-matching/c/rabin_karp_demo.c
#include "string_matching.h"
#include <stdio.h>
#include <string.h>

int main(void) {
    const unsigned char text[]="acebbceeaabceedb", pattern[]="eeaab";
    SmMatches out={0};
    if (!sm_rabin_karp_all(text,strlen((const char *)text),
                           pattern,strlen((const char *)pattern),5,113,&out)) return 1;
    printf("matches:");
    for (size_t i=0;i<out.count;i++) printf("%s%zu", i?",":" ", out.data[i]);
    putchar('\n');
    sm_matches_destroy(&out);
    return 0;
}

File: code/09-string-matching/python/rabin_karp_demo.py
from string_matching import rabin_karp

if __name__ == "__main__":
    text, pattern = "acebbceeaabceedb", "eeaab"
    matches = rabin_karp(text, pattern, 5, 113)
    print("matches:", ",".join(str(s) for s in matches))
